packages/patternfly-4/react-core/src/components/Dropdown/DropdownItem.js

```diff
@@ -25,41 +28,65 @@ const defaultProps = {
   isHovered: false,
   component: 'a',
   isDisabled: false,
-  href: '#'
+  href: '#',
+  onSelect: Function.prototype,
+  onClick: Function.prototype
 };
 
-const DropdownItem = ({ className, children, isHovered, component: Component, isDisabled, ...props }) => {
-  const additionalProps = props;
-  if (Component === 'a') {
-    additionalProps['aria-disabled'] = isDisabled;
-    additionalProps.tabIndex = isDisabled ? -1 : additionalProps.tabIndex;
-  } else if (Component === 'button') {
-    additionalProps.disabled = isDisabled;
-  }
+class DropdownItem extends React.Component {
```

Comment:
Out of curiosity, what is the reasoning behind switching to a class based component?  In any case, if we decide to keep this as a class, we can switch the context stuff over to the [contextType syntax](https://reactjs.org/docs/context.html#classcontexttype)